### PR TITLE
Switch to @automattic/eslint-plugin-wpvip

### PR DESCRIPTION
**Pull request by sjinks**
This PR is a follow-up to #3135. It migrates the rest of the codebase to `@automattic/eslint-plugin-wpvip`.

Most of the modifications to `.ts` files are automated fixes by ESLint.

### CODE REVIEW

File: __tests__/e2e/lib/pages/published-page-page.ts [hunk 1]
```diff
@@ -33,10 +33,12 @@ export class PublishedPagePage {
         // If text isn't found the first time, reload and check again up to 2 more times.
         let postTry = 0;
         while ( postTry < 3 ) {
+            // eslint-disable-next-line no-await-in-loop
             await this.page.waitForSelector( selectors.entryTitle );
-            if ( await this.page.locator( selectors.pageText( text ) ) ) {
+            if ( this.page.locator( selectors.pageText( text ) ) ) {
```
Comment (sjinks) [inline __tests__/e2e/lib/pages/published-page-page.ts:38]: @bsessions85 `this.page.locator()` returns a `Locator` and not a `Promise`; therefore, I removed `await`. According to the `.d.ts` file (and the [documentation](https://playwright.dev/docs/api/class-page#page-locator)), this method cannot return a falsy value. Thus, as far as I understand, this condition is always going to be `true`.

Does it make sense to await for `locator.count()` instead?

Could you please take a look?

File: __tests__/e2e/lib/pages/published-post-page.ts [hunk 2]
```diff
@@ -33,10 +33,12 @@ export class PublishedPostPage {
         // If text isn't found the first time, reload and check again up to 2 more times.
         let pageTry = 0;
         while ( pageTry < 3 ) {
+            // eslint-disable-next-line no-await-in-loop
             await this.page.waitForSelector( selectors.entryTitle );
-            if ( await this.page.locator( selectors.postText( text ) ) ) {
+            if ( this.page.locator( selectors.postText( text ) ) ) {
```
Comment (sjinks) [inline __tests__/e2e/lib/pages/published-post-page.ts:38]: @bsessions85 same as above

File: __tests__/e2e/lib/pages/published-page-page.ts (hunk 1 above)
Comment (bsessions85) [reply __tests__/e2e/lib/pages/published-page-page.ts:38]: Oh good question, totally missed that.
lets await for `locator.isVisible()`

File: __tests__/e2e/lib/pages/published-post-page.ts (hunk 2 above)
Comment (bsessions85) [reply __tests__/e2e/lib/pages/published-post-page.ts:38]: lets await for locator.isVisible()

File: __tests__/e2e/lib/pages/published-page-page.ts (hunk 1 above)
Comment (sjinks) [reply __tests__/e2e/lib/pages/published-page-page.ts]: Done

File: __tests__/e2e/lib/pages/published-post-page.ts (hunk 2 above)
Comment (sjinks) [reply __tests__/e2e/lib/pages/published-post-page.ts]: Done

File: __tests__/e2e/lib/pages/published-page-page.ts (hunk 1 above)
Comment (sjinks) [reply __tests__/e2e/lib/pages/published-page-page.ts]: This caused a strict mode violation:

```
    locator.isVisible: Error: strict mode violation: "text=8 Sad Badgers Slap Briefly" resolved to 2 elements:
        1) <a aria-current="page" class="wp-block-pages-list__…>8 Sad Badgers Slap Briefly</a> aka playwright.$("a:has-text("8 Sad Badgers Slap Briefly")")
        2) <h1 class="alignwide wp-block-post-title">8 Sad Badgers Slap Briefly</h1> aka playwright.$("h1:has-text("8 Sad Badgers Slap Briefly")")
```

I replaced `locator.iosVisible()` with `locator.first().isVisible()`.

**Review by bsessions85 — APPROVED** [review]

**Pull request merged**